mesonbuild/interpreter/interpreter.py

```diff
@@ -1493,19 +1493,17 @@ def find_program_fallback(self, fallback, args, required, extra_info):
     @FeatureNewKwargs('find_program', '0.49.0', ['disabler'])
     @disablerIfNotFound
     @permittedKwargs({'required', 'native', 'version', 'dirs'})
-    def func_find_program(self, node, args, kwargs) -> T.Union['build.Executable', ExternalProgram, 'OverrideProgram']:
-        if not args:
-            raise InterpreterException('No program name specified.')
-
+    @typed_pos_args('find_program', varargs=(str, mesonlib.File), min_varargs=1)
```

Comment:
Why do we allow mesonlib.File here? It was needed before all callers of self.func_find_program got fixed to use the impl, but, we do not document that this should work:

```meson
prog = files('foo.py')
prog = find_program(prog)
```

And it's super pointless to allow it. Anywhere that should transparently convert a files object into a found program, which I think literally just means configure_file, can upgrade that behind the scenes at time-of-use.

Reply:
Although I guess the correct solution here is to only allow configure_file outputs, which could be done by simply subclassing mesonlib.File with no changes? Maybe?